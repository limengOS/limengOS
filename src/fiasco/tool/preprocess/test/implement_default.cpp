INTERFACE:

class Test
{
public:
  void warn_func1();
  void warn_func2();
  void func1();
  void func2();
  void defl();

  // skip this as the test cannot deal with errors
  // void err_func1();
};

IMPLEMENTATION:

IMPLEMENT_DEFAULT void Test::warn_func1() { /* this is the default */ }
IMPLEMENT void Test::warn_func1()         { /* this is the override */ }

IMPLEMENT void Test::warn_func2()         { /* this is the override */ }
IMPLEMENT_DEFAULT void Test::warn_func2() { /* this is the default */ }

IMPLEMENT_DEFAULT void Test::func1()  { /* this is the default */ }
IMPLEMENT_OVERRIDE void Test::func1() { /* this is the override */ }

IMPLEMENT_OVERRIDE void Test::func2() { /* this is the override */ }
IMPLEMENT_DEFAULT void Test::func2()  { /* this is the default */ }

IMPLEMENT_DEFAULT void Test::defl()  { /* this is the default */ }

// skip this as the test cannot deal with errors
// IMPLEMENT_OVERRIDE void Test::err_func1() { /* this is the override */ }
