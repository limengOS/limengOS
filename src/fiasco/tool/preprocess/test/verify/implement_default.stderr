implement_default.cpp:19: warning: Function Test::warn_func1 overrides an IMPLEMENT_DEFAULT function but does not use IMPLEMENT_OVERRIDE
implement_default.cpp:20: warning: Function Test::warn_func2 overrides an IMPLEMENT_DEFAULT function but does not use IMPLEMENT_OVERRIDE
warning: found 2 warning(s)
